tests/commonlib/utils.py

```diff
@@ -110,3 +112,43 @@ def resource_identifier(resource):
             return dict_contains(body, dict(resource))
 
     return resource_identifier
+
+
+def wait_for_cycle_completion(elastic_client, nodes: list) -> bool:
+    """
+    Wait for all agents to finish sending findings to ES.
+    Done by waiting for all agents to send at least a single finding in the second cycle,
+    by that we verify that the first cycle is completed.
+    @param elastic_client: ES client
+    @param nodes: nodes list
+    @return: true if all agents finished sending findings in the configured timeout
+    """
+    required_cycles = 2
+    start_time = time.time()
+    prev_cycle_id = ""
+    curr_cycle_id = ""
+    active_agents = 0
+    num_cycles = 0
+
+    while num_cycles < required_cycles:
```

Comment:
@uri-weisman, num_cycles is iterated only if new cycle_id is retrieved. In case cloudbeat stops sending findings this will lead to infinite loop.

Reply:
Do we have a timeout per test?
Nevertheless, I can add a timeout to this function.

Reply:
currently test framework does not support timeouts per tests.

Reply:
I've added a timeout to the function